src/storage/plugins/CurlStorage.cxx

```diff
@@ -493,24 +516,31 @@ class HttpListDirectoryOperation final : public PropfindOperation {
 	 */
 	gcc_pure
 	StringView HrefToEscapedName(const char *href) const noexcept {
+		StringView relative_path;
 		StringView path = uri_get_path(href);
 		if (path == nullptr)
 			return nullptr;
 
 		/* kludge: ignoring case in this comparison to avoid
 		   false negatives if the web server uses a different
 		   case */
-		path = StringAfterPrefixIgnoreCase(path, base_path.c_str());
-		if (path == nullptr || path.empty())
+		relative_path = StringAfterPrefixIgnoreCase(path, base_path.c_str());
+		if (relative_path == nullptr || relative_path.empty()) {
```

Comment:
at first I wanted to use `uri_is_relative` on `path` but I didn't find how to convert a StringView to a c_str as it doesn't have a c_str method...